core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1765,6 +1765,7 @@ private[spark] object Utils extends Logging {
    */
   def getIteratorZipWithIndex[T](iterator: Iterator[T], startIndex: Long): Iterator[(T, Long)] = {
     new Iterator[(T, Long)] {
+      require(startIndex > 0, "startIndex should be > 0.")
```

Comment:
In the following line, you do `var index: Long = startIndex - 1L`. If it is == 0, then this line is -1L.